knowledge-base/treelist-enable-checkbox-selection-only-edit-mode.md

```diff
@@ -0,0 +1,354 @@
+2. Bind the [`Visible` parameter]({%slug treelist-columns-checkbox%}#parameters) of the `TreeListCheckboxColumn` to a variable. This allows toggling its visibility based on whether the TreeList is in edit mode.
+3. Handle the CUD (Create, Update, Delete) events to manage the visibility of the `TreeListCheckboxColumn`.
+    - For [`Inline`]({%slug treelist-editing-inline%}) and [`Popup`]({%slug treelist-editing-popup%}) edit modes, this approach works directly. In the [`Incell` edit mode the events flow is a bit different]({%slug treelist-editing-incell%}), so you may need to adjust the logic.
+
+Below is an example implementation that toggles the visibility of the Checkbox Column:
+
+```CSHTML
+@using System.ComponentModel.DataAnnotations @* for the validation attributes *@
+
+<TelerikTreeList Data="@Data"
+                 ItemsField="@(nameof(Employee.DirectReports))"
+                 EditMode="@TreeListEditMode.Inline"
+                 OnUpdate="@UpdateItem"
+                 OnDelete="@DeleteItem"
+                 OnAdd="@AddItem"
+                 OnCreate="@CreateItem"
+                 OnEdit="@OnEditHandler"
+                 OnCancel="@OnCancelHandler"
+                 SelectionMode="@TreeListSelectionMode.Multiple"
+                 @bind-SelectedItems="SelectedItems"
+                 Pageable="true">
+    <TreeListToolBarTemplate>
+        <TreeListCommandButton Command="Add" Icon="@SvgIcon.Plus">Add</TreeListCommandButton>
+    </TreeListToolBarTemplate>
+    <TreeListColumns>
+        <TreeListCheckboxColumn Visible="@SelectionEnabled" CheckBoxOnlySelection="true"/>
+
+        <TreeListColumn Field="Name" Expandable="true" Width="320px" />
+        <TreeListColumn Field="Id" Editable="false" Width="120px" />
+        <TreeListColumn Field="EmailAddress" Width="220px" />
+        <TreeListColumn Field="HireDate" Width="220px" />
+
+        <TreeListCommandColumn Width="280px">
+            <TreeListCommandButton Command="Add" Icon="@SvgIcon.Plus">Add Child</TreeListCommandButton>
+            <TreeListCommandButton Command="Edit" Icon="@SvgIcon.Pencil">Edit</TreeListCommandButton>
+            <TreeListCommandButton Command="Delete" Icon="@SvgIcon.Trash">Delete</TreeListCommandButton>
+            <TreeListCommandButton Command="Save" Icon="@SvgIcon.Save" ShowInEdit="true">Update</TreeListCommandButton>
+            <TreeListCommandButton Command="Cancel" Icon="@SvgIcon.Cancel" ShowInEdit="true">Cancel</TreeListCommandButton>
+        </TreeListCommandColumn>
+
+    </TreeListColumns>
+</TelerikTreeList>
+
+
+@code {
+    private List<Employee> Data { get; set; }
+    private IEnumerable<Employee> SelectedItems { get; set; } = new List<Employee>();
+
+    private bool SelectionEnabled { get; set; }    
+    private bool ItemAlreadySelected { get; set; }    
+
+    // Sample CUD operations for the local data
```

Comment:
Some comments of this sort are unnecessary, as the method names are self-evident. You can replace the necessary comments with `#region` if you like.